Pipfile

```diff
@@ -18,9 +18,9 @@ black = "*"
 build = "*"
 faker = "*"
 flake8 = "*"
-mcap = "*"
```

Comment:
What makes these dev-packages vs packages? If these are required dependencies?

Reply:
The intention of `dev-` vs. `packages` really isn't clear to me - https://stackoverflow.com/questions/53452152/what-is-pipenv-dev-packages-section-for says that dev packages are for "packages that don't need to be installed on the user machine", but there's nothing in the official docs to explain this.

The way we're using these sections right now, you need both `packages` and `dev-packages` to run tests. I guess in a ideal world the only thing under `packages` would be the `foxglove-data-platform` package itself as an editable install, but even that seems hacky - Pipenv does not have perfect support for editable installs.

Reply:
In nodejs package.json land the `devDependencies` are installed when you are working on the package locally (i.e. run install in the package repo). They are not installed when that package is used as a dependency in other packages. Whereas the `dependencies` are installed when that package is a dependency of other packages. If this is also true in python then it seems like these need to be dependencies if `py-data-platform` uses them directly.

Reply:
This isn't true in python - when installing a package only the dependencies in `setup.cfg` are considered (or setup.py if you're using that, we're not).

Reply:
py-data-platform uses these packages optionally, they're not required and you won't use them if you don't try to decode ros or protobuf messages.

Reply:
We can consider adding the mcap packages as an optional dependency that you install with `pip install foxglove-data-platform[mcap,protobuf,ros1]`, but i would want to do this in a separate PR. Also, the dependency story is a mess with ros1, it's basically impossible to install as a dependency automatically since it depends on a `genpy` from a non-default package index.

Reply:
https://github.com/foxglove/py-data-platform/issues/49